framework/src/play/src/main/scala/play/api/Configuration.scala

```diff
@@ -37,8 +41,9 @@ object Configuration {
         devSettings.get("config.file").orElse(Option(System.getProperty("config.file")))
           .map(f => new File(f)).getOrElse(new File(appPath, "conf/application.conf"))
       }
+      val playOverridesConfig: Config = getPlayOverridesConfig(Thread.currentThread().getContextClassLoader())
```

Comment:
My bad, I was looking at a wrong tag and the code in `play.api.Configuration` is quite different in Play 2.3 vs Play 2.4. So, it looks like in Play 2.3 we actually didn't have any facility to override configuration in `reference.conf`, so please disregard my above comment.